docs(router): Deprecate canLoad guards in favor of canMatch (#48180)
angular/angular: aio/content/guide/router-tutorial-toh.md

In `app.module.ts`, remove the `AdminModule` import statement from the top of the file and remove the `AdminModule` from the NgModule's `imports` array.

<a id="can-match-guard"></a>

### `canMatch`: guarding unauthorized access of feature modules

You're already protecting the `AdminModule` with a `canActivate` guard that prevents unauthorized users from accessing the admin feature area.
It redirects to the login page if the user is not authorized.

But the router is still loading the `AdminModule` even if the user can't visit any of its components.
Ideally, you'd only load the `AdminModule` if the user is logged in.

Add a `canMatch` guard that only matches the `AdminModule` once the user is logged in *and* attempts to access the admin feature area.

> A `canMatch` guard will control whether the `Router` even attempts to match a `Route`. This allows you to have

Reviewer: ```suggestion
A `canMatch` guard controls whether the `Router` attempts to match a `Route`. This lets you have
```

> multiple `Route` configurations which share the same `path` but are matched based on different conditions. In addition, this approach

Reviewer: ```suggestion
multiple `Route` configurations that share the same `path` but are matched based on different conditions. This approach
```

> can allow the `Router` to match the wildcard `Route` instead.

Reviewer: ```suggestion
allows the `Router` to match the wildcard `Route` instead.
```
But, I'm not sure if `instead` should be `as well`.

> The existing `authGuard` already has the essential logic to support the `canMatch` guard.

Reviewer: ```suggestion
The existing `authGuard` contains the logic to support the `canMatch` guard.
```

> All you need to do is add the `authGuard` to the `canMatch` array property for the `admin` route.

Reviewer: ```suggestion
Finally, add the `authGuard` to the `canMatch` array property for the `admin` route.
```
